releases/release-8.4.0.md

```diff
@@ -0,0 +1,255 @@
+summary: Learn about the new features, compatibility changes, improvements, and bug fixes in TiDB 8.4.0.
+---
+
+# TiDB 8.4.0 Release Notes
+
+Release date: xx xx, 2024
+
+TiDB version: 8.4.0
+
+Quick access: [Quick start](https://docs.pingcap.com/tidb/v8.4/quick-start-with-tidb)
+
+8.4.0 introduces the following key features and improvements:
+
+## Feature details
+
+### Scalability
+
+* Feature summary [#issue-number](issue-link) @[pr-auorthor-id](author-link)
+
+    Feature descriptions (including what the feature is, why it is valuable for users, and how to use this feature generally)
+
+    For more information, see [documentation](doc-link).
+
+### Performance
+
+* Feature summary [#issue-number](issue-link) @[pr-auorthor-id](author-link)
+
+    Feature descriptions (including what the feature is, why it is valuable for users, and how to use this feature generally)
+
+    For more information, see [documentation](doc-link).
+
+* The performance of batch user creation and password changes has been improved by hundreds of times [#55604](https://github.com/pingcap/tidb/pull/55604) @[wjhuang2016](https://github.com/wjhuang2016) **tw@hfxsd** <!--1941-->
+
+    In SaaS scenarios, you might need to batch-create a large number of users, rotate passwords periodically, and complete these tasks within a specific time window. Starting from v8.4.0, the performance of batch user creation and password rotation has been significantly improved. Additionally, you can further enhance performance by increasing concurrency through a higher number of session connections, which greatly reduces execution time for these operations.
+
+* Partitioned tables support global indexes (GA) [#45133](https://github.com/pingcap/tidb/issues/45133) @[mjonss](https://github.com/mjonss) @[Defined2014](https://github.com/Defined2014) @[jiyfhust](https://github.com/jiyfhust) @[L-maple](https://github.com/L-maple)
+
+    In previous versions of partitioned tables, some limitations exist because global indexes are not supported. For example, the unique key must use every column in the table's partitioning expression. If the query condition does not use the partitioning key, the query will scan all partitions, resulting in poor performance. Starting from v7.6.0, the system variable [`tidb_enable_global_index`](/system-variables.md#tidb_enable_global_index-new-in-v760) is introduced to enable the global index feature. But this feature was under development at that time and it is not recommended to enable it.
+
+    Starting with v8.3.0, the global index feature is released as an experimental feature. You can explicitly create a global index for a partitioned table with the keyword `Global` to remove the restriction that the unique key must use every column in the table's partitioning expression, to meet flexible business needs. Global indexes also enhance the performance of queries that do not include partition keys.
+
+    In v8.4.0, this feature becomes generally available (GA).
```

Comment:
```suggestion
    In v8.4.0, this feature becomes generally available (GA). You can directly use the keyword `GLOBAL` to create a global index, instead of setting the system variable [`tidb_enable_global_index`](/system-variables.md#tidb_enable_global_index-new-in-v760) to enable the global index feature.
```